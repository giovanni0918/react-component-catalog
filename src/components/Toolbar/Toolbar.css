.Toolbar {
    box-sizing: border-box;
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);
    width: 100%;
    height: 3.5rem;
    margin: 0;
    padding: 0;
    border: none;
    border-radius: 2px;
    background-color: dodgerblue;
    color: ghostwhite;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.2rem;
    font-weight: 600;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
}

.Toolbar a {
    flex: 1;
    text-align: center;
    text-decoration: none;
    color: inherit;
    will-change: transform;
    transform: scale(1.0);
    transition: transform 0.2s ease-in-out;
}

.Toolbar a:hover,
.Toolbar a:focus {
    text-decoration: underline;
    transform: scale(1.05);
}

.Toolbar--purple {
    background-color: #9C27B0;
    color: white;
}

.Toolbar--deep-purple {
    background-color: #673AB7;
    color: white;
}

.Toolbar--indigo {
    background-color: #3F51B5;
    color: white;
}

.Toolbar--teal {
    background-color: #009688;
    color: white;
}

.Toolbar--green {
    background-color: #4CAF50;
    color: black;
}

.Toolbar--primary {
    background-color: #03A9F4;
    color: black;
}

.Toolbar--info {
    background-color: powderblue;
    color: black;
}

.Toolbar--success {
    background-color: springgreen;
    color: black;
}

.Toolbar--warning {
    background-color: lightgoldenrodyellow;
    color: black;
}

.Toolbar--danger {
    background-color: palevioletred;
    color: black;
}