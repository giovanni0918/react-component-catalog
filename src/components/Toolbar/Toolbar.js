// @flow
import React from 'react';
import styles from './Toolbar.css';

type Props = { className: string, textContent: string, homeURL: string };

const Toolbar = ({
  className,
  textContent,
  homeURL
}: Props) => (
    <header className={className}>
      <a href={homeURL}>
        {textContent}
      </a>
    </header>
  );

export default Toolbar;