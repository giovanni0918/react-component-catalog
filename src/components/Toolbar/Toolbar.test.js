import React from 'react';
import Toolbar from './Toolbar';
import renderer from 'react-test-renderer';

const SampleToolbar = (
  <Toolbar
    className='Toolbar Toolbar--deep-purple'
    textContent='React Component Catalog'
    homeURL='#'
  />
);

test('It should always render a header', () => {
  const component = renderer.create(SampleToolbar);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('It should only contain children of type=a', () => {
  const component = renderer.create(SampleToolbar);
  let tree = component.toJSON();
  const isLink = (nodeChild) => nodeChild.type === 'a';
  expect(tree.children.every(isLink)).toBe(true);
});