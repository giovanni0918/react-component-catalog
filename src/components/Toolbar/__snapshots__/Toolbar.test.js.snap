// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should always render a header 1`] = `
<header
  className="Toolbar Toolbar--deep-purple"
>
  <a
    href="#"
  >
    React Component Catalog
  </a>
</header>
`;
